.github/actions/valgrind_run/action.yml

```diff
@@ -9,3 +9,10 @@ runs:
         valgrind --leak-check=full --error-exitcode=1 ./leaks_check
       shell: bash
       working-directory: ./tests/ndarrays
+    
+    - name: Test with valgrind for memory leaks in stc_containers
+      run: |
+        pyccel --language=c --flags="-g -O0" leaks_check.py
+        valgrind --leak-check=full --error-exitcode=1 ./leaks_check
```

Comment:
Maybe we should also double check the Fortran implementation

Reply:
Just checked that! turns out there are memory issues with `Fortran`. Shall I create an issue for that?

Reply:
Yes please

Reply:
Hopefully I guess I should be able to reuse most of this to fix the Fortran problem

Reply:
The only issue is about allocation for list comprehensions. Fortran does not allocate for the list before populating it. I don't think this issue is relevant in this context.

Reply:
Sounds like something that can be fixed using the `resize` attribute :+1:

Reply:
I created an issue for it #2036